docs/codegen.md

```diff
@@ -56,3 +55,63 @@ of two lines.
 | offset  | \<int\>         | This number will be added to the output if an address was found. (Negative offsets are valid) Default: `0`.|
 | type    | \<type name\>   | The type that the signature getter will output. It converts the value to this type with a reinterpret_cast. Default: `uintptr_t`.|
 | crucial | \<boolean\>     | Whether or not this signature is absolutely required. If true execution will halt after validation finds out that it was not found. Default `false`.|
+
+# Enum
+
+If type is `enum`
+
+## Definition
+
+```yaml
+enums:
+    - name: EnumTypeName
+      options:
+          - name: option 1
+          - name: option 2
+          - name: option 3
+          - name: another option
+            value: 5
+          - name: yet another option
```

Comment:
This seems like dangerous design. It requires people to be aware of the implementation detail that enums without values depend on the value of the previous enum. Your implementation does ensure unspecified values will still be unique, which is good.

I guess you could argue if the value isn't specified then it isn't important. What are these enums for?

Reply:
This exactly mirrors the C implementation of enums. Most enums have a starting value that affects the rest.

Reply:
To chime in here, as someone who's not familiar with C enums it's puzzling to me why one of these has a value and the others don't.

Reply:
When generated into CPP form they are explicitly numbered. Good standards usually makes it so that people only do this kind of thing at the start. But I was trying to knock out two examples in one here I guess

Reply:
Maybe documenting it a little better here will help?

Reply:
I think a more salient question is what do you plan on doing with these enums? Do the underlying values ever actually matter?

Reply:
For most they do actually. These are for the interaction between Vulpes and Halo. So, I know what you're going to say, it should be explicitly defined then.

Reply:
There is the odd few cases where it is one of our own though. But for netcode and cross version compatibility we might want to explicitly define those too.

Reply:
Sounds like we can reduce complexity by always requiring a value.